### ISSUE TITLE
Make strip prefix guessing less magic

### ISSUE THREAD

**Issue by nevir** [nevir/groc#13]
The current implementation is confusing

**Comment by SystemParadox** [comment]
Agreed. I am still trying to work out why it is stripping and how to stop it stripping when I don't want it to...
